sc/shared/src/test/scala/sigma/LanguageSpecificationV6.scala

```diff
@@ -1611,4 +1611,94 @@ class LanguageSpecificationV6 extends LanguageSpecificationBase { suite =>
 
   }
 
+  property("Coll.reverse") {
+    val f = newFeature[Coll[Int], Coll[Int]](
+      { (xs: Coll[Int]) => xs.reverse },
+      """{(xs: Coll[Int]) => xs.reverse }""".stripMargin,
+      sinceVersion = VersionContext.V6SoftForkVersion
+    )
+
+    verifyCases(
+      Seq(
+        Coll(1, 2) -> Expected(ExpectedResult(Success(Coll(2, 1)), None)),
+        Coll[Int]() -> Expected(ExpectedResult(Success(Coll[Int]()), None))
+      ),
+      f
+    )
+  }
+
+  property("Coll.distinct") {
+    val f = newFeature[Coll[Int], Coll[Int]](
+      { (xs: Coll[Int]) => xs.distinct },
+      """{(xs: Coll[Int]) => xs.distinct }""".stripMargin,
+      sinceVersion = VersionContext.V6SoftForkVersion
+    )
+
+    verifyCases(
+      Seq(
+        Coll(1, 2) -> Expected(ExpectedResult(Success(Coll(1, 2)), None)),
+        Coll(1, 1, 2) -> Expected(ExpectedResult(Success(Coll(1, 2)), None)),
+        Coll(1, 2, 2) -> Expected(ExpectedResult(Success(Coll(1, 2)), None)),
+        Coll[Int]() -> Expected(ExpectedResult(Success(Coll[Int]()), None))
+      ),
+      f
+    )
+  }
+
+  property("Coll.startsWith") {
+    val f = newFeature[(Coll[Int], Coll[Int]), Boolean](
+      { (xs: (Coll[Int], Coll[Int])) => xs._1.startsWith(xs._2) },
+      """{(xs: (Coll[Int], Coll[Int])) => xs._1.startsWith(xs._2) }""".stripMargin,
+      sinceVersion = VersionContext.V6SoftForkVersion
+    )
+
+    verifyCases(
+      Seq(
+        (Coll(1, 2, 3), Coll(1, 2)) -> Expected(ExpectedResult(Success(true), None)),
+        (Coll(1, 2, 3), Coll(1, 2, 3)) -> Expected(ExpectedResult(Success(true), None)),
+        (Coll(1, 2, 3), Coll(1, 2, 3, 4)) -> Expected(ExpectedResult(Success(false), None)),
+        (Coll[Int](), Coll[Int]()) -> Expected(ExpectedResult(Success(true), None))
```

Comment:
Not covered cases:
- Coll(1, 2, 3), Coll(1, 2, 4)
- Coll[Int](1, 2), Coll[Int]()

Reply:
added